src/Chainweb/Miner/Miners.hs

```diff
@@ -93,6 +95,20 @@ localTest lf v m cdb gen miners = runForever lf "Chainweb.Miner.Miners.localTest
     work :: BlockHeader -> IO BlockHeader
     work bh = MWC.geometric1 t gen >>= threadDelay >> pure bh
 
+-- | A miner that grabs new blocks from mempool and discards them. Mempool
+-- pruning happens during new-block time, so we need to ask for a new block
+-- regularly to prune mempool.
+mempoolNoopMiner :: LogFunction -> ChainwebVersion -> Miner -> CutDb cas -> IO ()
+mempoolNoopMiner lf v m cdb = runForever lf "Chainweb.Miner.Miners.mempoolNoopMiner" loop
+  where
+    chains = toList $ chainIds v
+    pact = _webPactExecutionService . _webBlockPayloadStorePact $
+           view cutDbPayloadStore cdb
+    loop = do
+        c <- _cut cdb
+        mapM_ (\cid -> newWork (Suggestion cid) m pact c) chains
```

Comment:
Does `IO` guarantee execution of the `mapM_`?

Reply:
Yes